node_modules/gh-core/lib/server.js

```diff
@@ -61,6 +61,44 @@ var setUpServer = module.exports.setUpServer = function(port, config) {
     // trust the X-Forwarded-* headers
     app.enable('trust proxy');
 
+    // Store the app on which the request was made on the request object
+    app.use(function(req, res, next) {
+        // Pass in a global admin server object when the request was made
+        // on the global admin host
+        if (config.servers.adminHostname === req.hostname) {
+            var globalAdmin = {
+                'displayName': 'Global admin server',
+                'host': req.hostname,
+                'isGlobalAdmin': true
+            };
```

Comment:
Or maybe a Util instead of an API if we can't get access to a context. But the DAO would preferably not hide this kind of logic.